### DISCUSSION TITLE
Question about _automaticDataCollectionEnabled

### DISCUSSION THREAD

**Discussion by atfilipcc** [invertase/react-native-firebase#5493] [Q&A] [answered]
Hello community! For GDPR reasons I am trying to disable analytics tracking for some users by calling 
`await firebase.analytics().setAnalyticsCollectionEnabled(false);`

however, when logging the firebase.analytics() object after the call I get an object back which includes 
`"_automaticDataCollectionEnabled": true`

Are these unrelated, or is it supposed to return false when data collection is off, and is this a reliable way to check that no analytics data is being sent from the device?

**Comment by mikehardy** [comment, marked as answer]
Interesting, I can see how this can be confusing.

The `@react-native-firebase/app` module does all the firebase.json / native parsing and initial setup (required for GDPR compliance to be able to start disabled even though default is enabled) and it handles multiple modules that all may collect information.

Analytics is "analyticsCollectionEnabled" as a general key (with different casing and underscores etc as it flows through the system)

In-App-Messaging is "automaticDataCollectionEnabled" as a general key (with different casing and underscores etc as it flows through the system)

For each module in use you'll need to examine the toggles but if you are not including / using In-App-Messaging you don't need to worry about the _automaticDataCollectionEnabled key

**Reply by atfilipcc** [reply]
Thank you Mike, that clears it up!
